packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/EmailAccountSettings.page.tsx

```diff
@@ -144,107 +149,78 @@ export default function EmailAccountSettings({
-      }
-    });
-
-    postZimbraPlatformAccount(platformId, dataBody)
-      .then(() => {
-        goBack();
-        addSuccess(
-          <OsdsText
-            color={ODS_THEME_COLOR_INTENT.text}
-            size={ODS_THEME_TYPOGRAPHY_SIZE._100}
-            level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
-            hue={ODS_THEME_COLOR_HUE._500}
-          >
-            {t('zimbra_account_add_success_message')}
-          </OsdsText>,
-          true,
-        );
-      })
-      .catch(({ response }) => {
-        goBack();
-        addError(
-          <OsdsText
-            color={ODS_THEME_COLOR_INTENT.text}
-            size={ODS_THEME_TYPOGRAPHY_SIZE._100}
-            level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
-            hue={ODS_THEME_COLOR_HUE._500}
-          >
-            {t('zimbra_account_add_error_message', {
-              error: response.data.message,
-            })}
-          </OsdsText>,
-          true,
-        );
+  const { mutate: addOrEditEmailAccount, isPending: isSending } = useMutation({
+    mutationFn: (params: AccountBodyParamsType) => {
+      return editEmailAccountId
+        ? putZimbraPlatformAccount(platformId, editEmailAccountId, params)
+        : postZimbraPlatformAccount(platformId, params);
+    },
+    onSuccess: () => {
+      addSuccess(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editEmailAccountId
+              ? 'zimbra_account_edit_success_message'
+              : 'zimbra_account_add_success_message',
+          )}
+        </OsdsText>,
+        true,
+      );
+    },
+    onError: (error: ApiError) => {
+      addError(
+        <OsdsText
+          color={ODS_THEME_COLOR_INTENT.text}
+          size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+          level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+          hue={ODS_THEME_COLOR_HUE._500}
+        >
+          {t(
+            editEmailAccountId
+              ? 'zimbra_account_edit_error_message'
+              : 'zimbra_account_add_error_message',
+            {
+              error: error?.response?.data?.message,
+            },
+          )}
+        </OsdsText>,
+        true,
+      );
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: getZimbraPlatformAccountsQueryKey(platformId),
       });
-  };
+      goBack();
+    },
+  });
 
-  const handleModifyAccountClick = () => {
+  const handleSaveClick = () => {
     const {
       account: { value: account },
       domain: { value: domain },
     } = form;
-
     let dataBody = {
       email: `${account}@${domain}`,
     };
-
     Object.entries(form).forEach(([key, { value }]) => {
       if (
         ![
-          ...['account', 'domain'],
-          ...[form.password.value === '' ? 'password' : ''],
+          'account',
```

Comment:
do we store constants values like these in constants files instead?

Reply:
we do not at the moment